pymatviz/ptable.py

```diff
@@ -332,17 +334,35 @@ def __init__(
             colormap (str | Colormap | None): The colormap to use.
             plot_kwargs (dict): Additional keyword arguments to
                 pass to the plt.subplots function call.
+            hide_f_block: Hide f-block (Lanthanum and Actinium series). Defaults to
+                None, meaning hide if no data is provided for f-block elements.
         """
         # Get colormap
         self.cmap: Colormap = colormap
 
         # Preprocess data
         self.data: pd.DataFrame = data
 
+        if hide_f_block is None:
+            hide_f_block = bool(
+                {
+                    atom_num
+                    for atom_num in [*range(57, 72), *range(89, 104)]  # rare earths
+                    # check if data is present for f-block elements
+                    if (elem := Element.from_Z(atom_num).symbol) in self.data.index  # type: ignore[union-attr]
+                    and self.data.loc[elem, "Value"]  # type: ignore[union-attr]
```

Comment:
one more thing to change here: it's always a good to maintain [SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth) to avoid drift and so it's to change things in the future. the `"Value"` is hard-coded in many places in this file. would be better to add a [new key in init.py](https://github.com/danielyang59/pymatviz/blob/69ea94920082b814d682ed3b3ccda5999b018009/pymatviz/__init__.py#L91)

Reply:
Yes, as we're hard code the column names: https://github.com/janosh/pymatviz/blob/39fd71e5e105574ed1e8d1e3174f9ae4257f95be/pymatviz/ptable.py#L217-L254

Maybe it's good to define a standard name of both columns in case we need to change them in the future?